tests/TestCase/Database/DriverTest.php

```diff
@@ -0,0 +1,257 @@
+    {
+        $this->driver->_connection = 'connection';
+        $this->assertTrue($this->driver->isConnected());
+
+        $this->driver->_connection = null;
+        $this->assertFalse($this->driver->isConnected());
+    }
+
+    /**
+     * test autoQuoting().
+     *
+     * @return void
+     */
+    public function testAutoQuoting()
+    {
+        $this->assertFalse($this->driver->autoQuoting());
+
+        $this->driver->autoQuoting(true);
+        $this->assertTrue($this->driver->autoQuoting());
+
+        $this->assertTrue($this->driver->autoQuoting(true));
+        $this->assertFalse($this->driver->autoQuoting(false));
+
+        $this->assertTrue($this->driver->autoQuoting('string'));
+        $this->assertFalse($this->driver->autoQuoting('0'));
+
+        $this->assertTrue($this->driver->autoQuoting(1));
+        $this->assertFalse($this->driver->autoQuoting(0));
+    }
+
+    /**
+     * Test compileQuery().
+     *
+     * @return void
+     * @group new
+     */
+    public function testCompileQuery()
+    {
+        $compiler = $this->getMockBuilder(QueryCompiler::class)
+            ->setMethods(['compile'])
+            ->getMock();
+
+        $compiler
+            ->expects($this->once())
+            ->method('compile')
+            ->willReturn(true);
+
+        $driver = $this->getMockBuilder(Driver::class)
+            ->setMethods(['newCompiler', 'queryTranslator'])
+            ->getMockForAbstractClass();
+
+        $driver
+            ->expects($this->once())
+            ->method('newCompiler')
+            ->willReturn($compiler);
+
+        $driver
+            ->expects($this->once())
+            ->method('queryTranslator')
+            ->willReturn(function ($query) {
+                return $query;
+            });
+
+        $query = $this->getMockBuilder(Query::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $result = $driver->compileQuery($query, new ValueBinder);
+
+        $this->assertInternalType('array', $result);
+        $this->assertSame($query, $result[0]);
+        $this->assertTrue($result[1]);
+    }
+
+    /**
+     * Test newCompiler().
+     *
+     * @return void
+     */
+    public function testNewCompiler()
+    {
+        $this->assertInstanceOf(QueryCompiler::class, $this->driver->newCompiler());
+    }
+
+    /**
+     * Test __destruct().
+     *
+     * @return void
+     */
+    public function testDestructor()
+    {
+        $this->driver->_connection = true;
+        $this->driver->__destruct();
+
+        $this->assertNull($this->driver->_connection);
+    }
+
+    /**
+     * Data provider for testSchemaValue().
+     *
+     * @return array
+     */
+    public function schemaValueProvider()
+    {
+        return [
+            [null, 'NULL'],
+            [false, 'FALSE'],
+            [true, 'TRUE'],
+            [1, 1],
+            ['0', '0'],
+            ['42', '42'],
+            ['string', true]
```

Comment:
I think it might be better to separate that test case out. Right now it reads like string input = boolean output which is not what would happen in a real implementation.